gamestonk_terminal/cryptocurrency/defi/coindix_model.py

```diff
@@ -0,0 +1,130 @@
+    "eth",
+    "harmony",
+    "fuse",
+    "defichain",
+    "solana",
+    "optimism",
+]
+PROTOCOLS = [
+    "aave",
+    "acryptos",
+    "alpaca",
+    "anchor",
+    "autofarm",
+    "balancer",
+    "bancor",
+    "beefy",
+    "belt",
+    "compound",
+    "convex",
+    "cream",
+    "curve",
+    "defichain",
+    "geist",
+    "lido",
+    "liquity",
+    "mirror",
+    "pancakeswap",
+    "raydium",
+    "sushi",
+    "tarot",
+    "traderjoe",
+    "tulip",
+    "ubeswap",
+    "uniswap",
+    "venus",
+    "yearn",
+]
+VAULT_KINDS = ["lp", "single", "noimploss", "stable"]
+
+
+def _prepare_params(**kwargs) -> dict:
+    """Helper method, which handles preparation of parameters for requests to coindix api.
+
+    Parameters
+    ----------
+    kwargs: keyword arguments: chain, kind, protocol
+
+    Returns
+    -------
+    dict:
+        Prepared parameters for request
+    """
+
+    params = {"sort": "-apy", "tvl": "1m", "kind": "all"}
+    mapping = {"chain": CHAINS, "protocol": PROTOCOLS, "kind": VAULT_KINDS}
+    for key, value in kwargs.items():
+        category = mapping.get(key, [])
+        if value in category:
+            params.update({key: value})
+    return {k: v.lower() for k, v in params.items()}
+
+
+def get_defi_vaults(
+    chain: Optional[str] = None,
+    protocol: Optional[str] = None,
+    kind: Optional[str] = None,
+) -> pd.DataFrame:
+    """Get DeFi Vaults Information. DeFi Vaults are pools of funds with an assigned strategy which main goal is to
+    maximize returns of its crypto assets. [Source: https://coindix.com/]
+
+    Parameters
+    ----------
+    chain: str
+        Blockchain - one from list [
+            'ethereum', 'polygon', 'avalanche', 'bsc', 'terra', 'fantom',
+            'moonriver', 'celo', 'heco', 'okex', 'cronos', 'arbitrum', 'eth',
+            'harmony', 'fuse', 'defichain', 'solana', 'optimism'
+        ]
+    protocol: str
+        DeFi protocol - one from list: [
+            'aave', 'acryptos', 'alpaca', 'anchor', 'autofarm', 'balancer', 'bancor',
+            'beefy', 'belt', 'compound', 'convex', 'cream', 'curve', 'defichain', 'geist',
+            'lido', 'liquity', 'mirror', 'pancakeswap', 'raydium', 'sushi', 'tarot', 'traderjoe',
+            'tulip', 'ubeswap', 'uniswap', 'venus', 'yearn'
+        ]
+    kind: str
+        Kind/type of vault - one from list: ['lp','single','noimploss','stable']
+
+    Returns
+    -------
+    pd.DataFrame
+        Top 100 DeFi Vaults for given chain/protocol sorted by APY.
+    """
+
+    params = _prepare_params(chain=chain, protocol=protocol, kind=kind)
+    response = requests.get("https://apiv2.coindix.com/search", params=params)
+    if not 200 <= response.status_code < 300:
+        raise Exception(f"Coindix api exception: {response.text}")
+
+    try:
+        data = response.json()["data"]
+        if len(data) == 0:
+            return pd.DataFrame()
+        df = pd.DataFrame(data)[VAULTS_FILTERS].fillna("NA")
```

Comment:
What do you think about capitalizing the column names renaming apy to APY (%) and tvl to TVL ($)?

Reply:
Will do, but in the view level - this way user can do `-s apy` instead of `--s "APY (%)"` much better user experience

Reply:
oh, did not thought about that. can we map apy to APY (%) for this specific use case?